qa/vagrant/build.gradle

```diff
@@ -95,8 +115,33 @@ task stop {
   description 'Stop any tasks from tests that still may be running'
 }
 
+Set<String> getVersions() {
+  Node xml
+  new URL('http://repo1.maven.org/maven2/org/elasticsearch/elasticsearch/maven-metadata.xml').openStream().withStream { s ->
+    xml = new XmlParser().parse(s)
+  }
+  return new TreeSet<>(xml.versioning.versions.version.collect { it.text() }.findAll { it ==~ /2\.\d\.\d/ })
+}
+
+task updatePackagingTestUpgradeFromVersions {
+  Set<String> versions = getVersions()
+  new File(project.projectDir, 'versions').text = versions.join('\n') + '\n'
+}
+
+task verifyPackagingTestUpgradeFromVersions {
+  String maybeUpdateFromVersions = System.getProperty("tests.packaging.upgrade.from.versions", null)
```

Comment:
The way this is written now it happens at task construction time, not task execution time. So this is done every time gradle is started. If you wrap it in `doLast` then it'll wait. The same is true of the above task.